src/sqlancer/hive/gen/HiveTableGenerator.java

```diff
@@ -0,0 +1,125 @@
+        this.gen = new HiveExpressionGenerator(globalState).setColumns(columnsToBeAdded);
+    }
+
+    public static SQLQueryAdapter generate(HiveGlobalState globalState, String tableName) {
+        HiveTableGenerator generator = new HiveTableGenerator(globalState, tableName);
+        return generator.create();
+    }
+
+    private SQLQueryAdapter create() {
+        ExpectedErrors errors = new ExpectedErrors();
+
+        sb.append("CREATE ");
+        sb.append("TABLE ");
+        sb.append(globalState.getDatabaseName());
+        sb.append(".");
+        sb.append(tableName);
+        sb.append(" (");
+        for (int i = 0; i < Randomly.smallNumber() + 1; i++) {
+            if (i != 0) {
+                sb.append(", ");
+            }
+            appendColumn(i, errors);
+        }
+        sb.append(")");
+
+        // TODO: implement PARTITION BY clause
+        // TODO: implement CLUSTERED BY, SKEWED BY clauses
+        // TODO: implement ROW FORMAT and STORED AS clauses
+        // TODO: randomly add some predefined TABLEPROPERTIES
+        // TODO: implement CTAS (AS clause)
+
+        HiveErrors.addExpressionErrors(errors);
+        return new SQLQueryAdapter(sb.toString(), errors, true, false);
+    }
+
+    private void appendColumn(int columnId, ExpectedErrors errors) {
+        String columnName = DBMSCommon.createColumnName(columnId);
+        sb.append(columnName);
+        sb.append(" ");
+        HiveDataType randType = HiveSchema.HiveDataType.getRandomType();
+        sb.append(randType);
+        columnsToBeAdded.add(new HiveColumn(columnName, table, randType));
+        appendColumnConstraint(randType, errors);
+    }
+
+    private void appendColumnConstraint(HiveDataType type, ExpectedErrors errors) {
+        /* column_constraint_specification:
+             : [ PRIMARY KEY|UNIQUE|NOT NULL|DEFAULT [default_value]|CHECK  [check_expression] ENABLE|DISABLE NOVALIDATE RELY/NORELY ] 
+         */
+        if (Randomly.getBoolean()) {
+            // no column constraint
+            return;
+        }
+
+        ColumnConstraints constraint = Randomly.fromOptions(ColumnConstraints.values());
+        switch (constraint) {
+            case PRIMARY_KEY_DISABLE:
+                if (allowPrimaryKey && !setPrimaryKey) {
+                    sb.append(" PRIMARY KEY DISABLE");
+                    setPrimaryKey = true;
+                }
+                break;
+            case UNIQUE_DISABLE:
+                sb.append(" UNIQUE DISABLE");
+                break;
+            case NOT_NULL:
+                sb.append(" NOT NULL");
+                break;
+            case DEFAULT:
+                // TODO: add errors
+                sb.append(" DEFAULT (");
+                sb.append(HiveToStringVisitor.asString(gen.generateConstant()));
+                sb.append(")");
+            case CHECK:
+                errors.add("Invalid Constraint syntax");
```

Comment:
I think we should always generate constraints with a valid syntax. Or is there a justified case here?